lib/cpp/src/thrift/server/TThreadPoolServer.cpp

```diff
@@ -146,9 +73,12 @@ void TThreadPoolServer::serve() {
       shared_ptr<TProcessor> processor = getProcessor(inputProtocol, outputProtocol, client);
 
       // Add to threadmanager pool
-      shared_ptr<TThreadPoolServer::Task> task(
-          new TThreadPoolServer::Task(*this, processor, inputProtocol, outputProtocol, client));
-      threadManager_->add(task, timeout_, taskExpiration_);
+      shared_ptr<TConnectedClient> pClient(
```

Comment:
I thought that was supposed to be used when making a shared pointer from within the code of an existing smart pointer, i.e. a class that is boost::enable_shared_from_this.  Wasn't aware it could be used in that manner.  I'll look into this and I can roll it into the subsequent call, I do this normally but was erring on the side of caution on the amount of fanciness.